x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/detection_rules_client/detection_rules_client.ts

```diff
@@ -31,30 +33,31 @@ import { importRule } from './methods/import_rule';
 import { withSecuritySpan } from '../../../../../utils/with_security_span';
 
 export const createDetectionRulesClient = (
```

Comment:
This is maybe nitpicking, but given that we're dealing with the `DetectionRulesClient` here, I strongly feel that the `actionsClient` should not be the first parameter of `createDetectionRulesClient`. Can we reordering it so?
```ts
export const createDetectionRulesClient = (
  rulesClient: RulesClient,
  actionsClient: ActionsClient,
  mlAuthz: MlAuthz
): IDetectionRulesClient => ({
  ...
```
Same goes with the arguments of all the methods used here. Instead of:
```ts
createCustomRule(actionsClient, rulesClient, args, mlAuthz);
```
This should be:
```ts
createCustomRule(rulesClient, args, actionsClient, mlAuthz);
```
The idea being, the most important dependency for interacting with rules is the `rulesClient`. Then, we need the arguments to create/modify/delete them. And only after that, additional dependencies like `actionsClient` and `mlAutzh`.

Reply:
@dhurley14 Just a heads up that I've refactored this builder function to accept arguments as an object instead of positional arguments [in my PR](https://github.com/elastic/kibana/pull/186988):

```ts
interface DetectionRulesClientParams {
  rulesClient: RulesClient;
  savedObjectsClient: SavedObjectsClientContract;
  mlAuthz: MlAuthz;
}

export const createDetectionRulesClient = ({
  rulesClient,
  mlAuthz,
  savedObjectsClient,
}: DetectionRulesClientParams): IDetectionRulesClient => {
```

I'd propose aligning here, so we don't end up with a lot of conflicts.

Reply:
This should be fixed now.